File: Lab5/VkontakteRecycleVIew/app/src/main/java/com/example/vkrv/MainActivity.java
package com.example.vkrv;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.os.Parcelable;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements PostsListAdapter.FragmentButtonListener,
PostsListAdapter.FragmentLikeListener, LikedListAdapter.FragmentLikeListener{
    private LockableViewPager lockableViewPager;
    private PagerAdapter pagerAdapter;
    Fragment fragment1 = new MainFragment();
    Fragment fragment2 = new LikeFragment();
    List<Fragment> list = new ArrayList<>();
    BottomNavigationView bottomNavigationView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        bottomNavigationView = findViewById(R.id.bn);
        list.add(fragment1);
        list.add(fragment2);
        lockableViewPager = findViewById(R.id.lvp);
        pagerAdapter = new PagerAdapter(getSupportFragmentManager(), list);
        lockableViewPager.setAdapter(pagerAdapter);
        lockableViewPager.setSwipable(false);

        bottomNavigationView.setOnNavigationItemSelectedListener(
                new BottomNavigationView.OnNavigationItemSelectedListener() {
                    @Override
                    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                        switch (item.getItemId()){
                            case R.id.main:
                                lockableViewPager.setCurrentItem(0, false);
                                getSupportActionBar().setTitle("Новости");

                                break;
                            case R.id.favs:
                                lockableViewPager.setCurrentItem(1, false);
                                getSupportActionBar().setTitle("Понравившиеся");

                                break;

                        }
                        return false;
                    }
                }
        );
    }

    @Override
    public void myClick(Posts posts, int option) {
        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.lvp);
        if (option ==1)
            ((LikeFragment)fragment).savePosts(posts);
        else
            ((LikeFragment)fragment).removePosts(posts);
    }

    @Override
    public void removeItemLike(Posts posts) {
        ((MainFragment)fragment1).removeLike(posts);
        ((LikeFragment)fragment2).removeLike(posts);

    }
}

File: LabsInKotlin/Lab5/app/src/main/java/com/example/lab5/MainActivity.kt
package com.example.lab5

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.google.android.material.bottomnavigation.BottomNavigationView
import java.util.*


class MainActivity : AppCompatActivity(), PostsListAdapter.FragmentButtonListener,
    LikedListAdapter.FragmentLikeListener {
    var fragment1: Fragment = MainFragment()
    var fragment2: Fragment = LikeFragment()
    private lateinit var pager: ViewPager
    private lateinit var pagerAdapter:PagerAdapter
    var list: MutableList<Fragment> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        list.add(fragment1)
        list.add(fragment2)
        pager = findViewById(R.id.pager)
        pagerAdapter = Pager(supportFragmentManager, list)
        pager.adapter = pagerAdapter
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bn)
        bottomNavigationView.setOnNavigationItemSelectedListener(navigationListener)
    }
    private val navigationListener =
            BottomNavigationView.OnNavigationItemSelectedListener {  item ->
                when (item.itemId) {
                    R.id.main -> {
                        pager.setCurrentItem(0, false)
                        supportActionBar!!.title = "Новости"
                        return@OnNavigationItemSelectedListener true

                    }
                    R.id.favs -> {
                        pager.setCurrentItem(1, false)
                        supportActionBar!!.title = "Закладки"
                        return@OnNavigationItemSelectedListener true
                    }
                }
                return@OnNavigationItemSelectedListener false

            }
    override fun removeItemLike(posts: Posts?) {
        (fragment1 as MainFragment).removeLike(posts)
        (fragment2 as LikeFragment).removeLike(posts)
    }


    override fun myClick(posts: Posts?, option: Int) {
        val fragment = supportFragmentManager.findFragmentById(R.id.pager)
        if(option == 1) posts?.let { (fragment as LikeFragment?)?.savePosts(it) }
        else (fragment as LikeFragment?)?.removePosts(Posts)
    }


}



